offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
+            , Pool { assetA: Asset { name: "DADA", value: fromInt 3000 }, assetB: Asset { name: "USDC", value: fromInt 1000 }, fee: 4 }
+            ]
+        , currentPool: Nothing
+        , activity: Swap
+        }
     , render
     , eval: H.mkEval $ H.defaultEval
+        { handleAction = handleAction
+        }
     }
   where
-  render _ =
-    HH.div_
-      [ HH.text "this is Home"
+  handleAction :: forall slots. Action -> H.HalogenM State Action slots o m Unit
+  handleAction = case _ of
+    ResetCurrentPool -> H.modify_ _ { currentPool = Nothing }
+    SetCurrentPool pool -> H.modify_ _ { currentPool = Just pool }
+    SetActivity activity -> H.modify_ _ { activity = activity }
+
+  render :: forall slots. State -> H.ComponentHTML Action slots m
+  render { pools, currentPool, activity } =
+    HH.section
+      [ mkClass "section my-6 pb-6" ]
+      $
+        [ HH.div
+            [ mkClass "columns is-mobile" ]
+            [ HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-left" ] [ HH.text "Pool" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Value" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Fee" ]
+                    ]
+                ]
+            , HH.div [ mkClass "column" ] [ HH.text " " ]
+            ]
+
+        ]
+      <> concat
+        ( renderPool
+            <$> pools
+        )
+    where
+    renderPool pool@(Pool p) =
+      [ HH.div
+          [ mkClass "pools-table-row columns is-mobile is-vcentered p-3 my-3"
+          , HE.onClick $ \_ ->
+              if currentPool == Just pool then
+                ResetCurrentPool
+              else
+                SetCurrentPool pool
+          ]
+
+          [ HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-left" ] [ HH.text $ show pool ]
+              ]
+          , HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-centered" ] [ HH.text $ show p.assetA <> " / " <> show p.assetB ]
+              ]
+          , HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-centered" ] [ HH.text $ show p.fee <> "ADA" ]
+              ]
+          , HH.div [ mkClass "column has-text-centered" ]
+              [ if currentPool == Just pool then mkIcon "angle-up" else mkIcon "angle-down"
```

Comment:
We are missing some accessibility text here for what these mean. I suggest a `title` but also a `aria-label` or similar (or a `<span class="srOnly">` for screen readers).

Reply:
sure thing, will add `<span class="srOnly">` to the icons

Reply:
It's easy to write :stuck_out_tongue: 

```css
.srOnly {
   overflow: hidden !important;
   position: absolute !important;
   width: 1px !important;
   height: 1px !important;
   padding: 0 !important;
   border: none !important;
   -webkit-clip-path: inset(50%) !important;
   clip-path: inset(50%) !important;
}
```

Reply:
after I apply the above to all icons, they disappear ...

Reply:
Ha. Nooo. You apply this to any labels you need. Icons aren't text so they don't tell you they mean.

Reply:
I tried to apply to text like `HH.span [ mkClass "sr-only" ] [ HH.text "Slippage: 4%" ]`, the text disappear, do you think this is a must-have for MVP?

Reply:
Disappears to you, the able-bodied user--but not to our machine and blind friends.

Reply:
These labels will also help future coders understand what the heck does "angle-up" mean